sdk/typescript/src/utils/gas-estimation.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { UnserializedSignableTransaction } from '../signers/txn-data-serializers/txn-data-serializer';
+import { Coin, SuiMoveObject } from '../types';
+
+export type TxKind = UnserializedSignableTransaction['kind'];
+type BudgetMap = typeof DEFAULT_GAS_BUDGET_PER_TX_TYPE;
+export type GasBudgetGuessParams<txKind extends TxKind> =
+  txKind extends keyof BudgetMap
+    ? BudgetMap[txKind] extends (...args: any[]) => any
+      ? Parameters<BudgetMap[txKind]> extends [boolean, ...infer Params]
+        ? Params
+        : []
+      : []
+    : [];
+
+export const MIN_GAS_BUDGET = 10;
+export const MAX_GAS_BUDGET = 1_000_000;
+
+const PAY_GAS_FEE_PER_COIN = 150;
```

Comment:
I assume this needs to be kept in sync with some backend value, can we document that?

Reply:
I guess, moved this from Coin so not sure why is this number, I will add a comment

Reply:
done